src/unit_tests/wazuh_db/test_wdb_parser.c

```diff
@@ -740,7 +741,479 @@ void test_wdb_parse_rootcheck_save_update_insert_success(void **state)
+
+    os_free(query);
+}
+
+void test_osinfo_set_error_no_scan_time(void **state) {
+    int ret = OS_INVALID;
+    test_struct_t *data  = (test_struct_t *)*state;
+    char *query = NULL;
+
+    os_strdup("set NULL|", query);
+
+    // wdb_parse_agents_set_sys_osinfo
+    expect_string(__wrap__mdebug1, formatted_msg, "Invalid OS info query syntax.");
+    expect_string(__wrap__mdebug2, formatted_msg, "OS info query: ");
+
+    ret = wdb_parse_osinfo(data->wdb, query, data->output);
+
+    assert_string_equal(data->output, "err Invalid OS info query syntax, near ''");
+    assert_int_equal(ret, OS_INVALID);
+
+    os_free(query);
+}
+
+void test_osinfo_set_error_no_hostname(void **state) {
+    int ret = OS_INVALID;
+    test_struct_t *data  = (test_struct_t *)*state;
+    char *query = NULL;
+
+    os_strdup("set scan_id|NULL|", query);
+
+    // wdb_parse_agents_set_sys_osinfo
+    expect_string(__wrap__mdebug1, formatted_msg, "Invalid OS info query syntax.");
+    expect_string(__wrap__mdebug2, formatted_msg, "OS info query: ");
+
+    ret = wdb_parse_osinfo(data->wdb, query, data->output);
+
+    assert_string_equal(data->output, "err Invalid OS info query syntax, near ''");
+    assert_int_equal(ret, OS_INVALID);
+
+    os_free(query);
+}
+
+void test_osinfo_set_error_no_architecture(void **state) {
+    int ret = OS_INVALID;
+    test_struct_t *data  = (test_struct_t *)*state;
+    char *query = NULL;
+
+    os_strdup("set scan_id|scan_time|NULL|", query);
+
+    // wdb_parse_agents_set_sys_osinfo
+    expect_string(__wrap__mdebug1, formatted_msg, "Invalid OS info query syntax.");
+    expect_string(__wrap__mdebug2, formatted_msg, "OS info query: ");
+
+    ret = wdb_parse_osinfo(data->wdb, query, data->output);
+
+    assert_string_equal(data->output, "err Invalid OS info query syntax, near ''");
+    assert_int_equal(ret, OS_INVALID);
+
+    os_free(query);
+}
+
+void test_osinfo_set_error_no_os_name(void **state) {
+    int ret = OS_INVALID;
+    test_struct_t *data  = (test_struct_t *)*state;
+    char *query = NULL;
+
+    os_strdup("set scan_id|scan_time|hostname|NULL|", query);
+
+    // wdb_parse_agents_set_sys_osinfo
+    expect_string(__wrap__mdebug1, formatted_msg, "Invalid OS info query syntax.");
+    expect_string(__wrap__mdebug2, formatted_msg, "OS info query: ");
+
+    ret = wdb_parse_osinfo(data->wdb, query, data->output);
+
+    assert_string_equal(data->output, "err Invalid OS info query syntax, near ''");
+    assert_int_equal(ret, OS_INVALID);
+
+    os_free(query);
+}
+
+void test_osinfo_set_error_no_os_version(void **state) {
+    int ret = OS_INVALID;
+    test_struct_t *data  = (test_struct_t *)*state;
+    char *query = NULL;
+
+    os_strdup("set scan_id|scan_time|hostname|architecture|NULL|", query);
+
+    // wdb_parse_agents_set_sys_osinfo
+    expect_string(__wrap__mdebug1, formatted_msg, "Invalid OS info query syntax.");
+    expect_string(__wrap__mdebug2, formatted_msg, "OS info query: ");
```

Comment:
We could analyze this error messages, currently they are empty

Reply:
This was improved as per the discussion that we had.